linker/shared/src/main/scala/org/scalajs/linker/checker/ClassDefChecker.scala

```diff
@@ -100,8 +100,17 @@ private final class ClassDefChecker(classDef: ClassDef, reporter: ErrorReporter)
   }
 
   private def checkKind()(implicit ctx: ErrorContext): Unit = {
-    if (isJLObject && classDef.kind != ClassKind.Class)
+    val className = classDef.name.name
+
+    if (isJLObject && classDef.kind != ClassKind.Class) {
       reportError("java.lang.Object must be a Class")
+    } else {
+      val isHijacked = BoxedClassToPrimType.contains(className)
```

Comment:
Hmm... Is `ir.Names.HijackedClasses` the source of truth for hijacked classes or `BoxedClassToPrimType`? Or do we use them interchangeably?

Reply:
Ah using `HijackedClasses` would be better here, indeed.

(They are both source of truth; if they're not in sync, something it's terribly wrong anyway.)